backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java

```diff
@@ -448,19 +449,27 @@ public static Throwable findRootCause(Throwable throwable) {
                 .addDetails(Any.pack(defaultInstance))
                 .build();
       } else {
-        LOGGER.error("Exception occurred: {}", e.getMessage());
-        StackTraceElement[] stack = e.getStackTrace();
-        int maxLines = (stack.length > 4) ? 5 : stack.length;
-        for (int n = 0; n < maxLines; n++) {
-          LOGGER.error(stack[n].toString());
-        }
         status =
             Status.newBuilder()
                 .setCode(Code.INTERNAL_VALUE)
                 .setMessage(ModelDBConstants.INTERNAL_ERROR)
                 .addDetails(Any.pack(defaultInstance))
                 .build();
       }
+      StackTraceElement[] stack = e.getStackTrace();
+      LOGGER.error("Stacktrace with {} elements for {}", stack.length, e.getMessage());
+      int n = 0;
+      boolean isLongStack = stack.length > STACKTRACE_LENGTH;
+      if (isLongStack) {
+        for (; n < STACKTRACE_LENGTH + 1; ++n) {
+          LOGGER.warn("{}: {}", n, stack[n].toString());
+        }
+      }
+      for (; n < stack.length; ++n) {
+        if (stack[n].getClassName().startsWith("ai.verta") || !isLongStack) {
+          LOGGER.warn("{}: {}", n, stack[n].toString());
+        }
+      }
```

Comment:
Unrelated? If yes, remove it.

Reply:
it's unrelated, but was very useful when testing this functionality. I have removed it, will create a separate PR.